### PR TITLE
Ports insectivorous lizards from TG

### PR DESCRIPTION
**Pull request by Pathid**
:cl:
* rscadd: Lizards eat cockroaches now. Player-controlled ones can also ventcrawl and hide like mice can.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/lizard.dm [hunk 1]
```diff
@@ -18,3 +18,50 @@
...
+/mob/living/simple_animal/hostile/lizard/CanAttack(var/atom/the_target)//Can we actually attack a possible target?
+	if(see_invisible < the_target.invisibility)//Target's invisible to us, forget it
+		return FALSE
+	if(is_type_in_list(the_target, edibles))
+		return TRUE
+	return FALSE
+
+/mob/living/simple_animal/hostile/lizard/AttackingTarget()
+	if(is_type_in_list(target, edibles)) //Makes sure player lizards only consume edibles.
+		visible_message("[name] consumes [target] in a single gulp", "<span class='notice'>You consume [target] in a single gulp</span>")
```
Comment (jknpj) [inline code/modules/mob/living/simple_animal/friendly/lizard.dm]: No sound?

**Comment by Aurx** [comment]
Why not bees?

**Comment by SonixApache** [comment]
Why would bees eat cockroaches and ventcrawl?

File: code/modules/mob/living/simple_animal/friendly/lizard.dm [hunk 2]
```diff
@@ -18,3 +18,50 @@
 	mob_property_flags = MOB_NO_PETRIFY //Can't get petrified (nethack references)
 	meat_type = /obj/item/weapon/reagent_containers/food/snacks/meat/animal/lizard
 	held_items = list()
+
+	stop_automated_movement_when_pulled = TRUE
+	environment_smash_flags = FALSE
+	density = FALSE
+	pass_flags = PASSTABLE | PASSMOB
+
+	var/static/list/edibles = list(/mob/living/simple_animal/cockroach) //Add bugs to this as they get added in
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/friendly/lizard.dm]: Add cockroach eggs to this

**Comment by Kurfursten** [comment]
Seems like there should be a lizard crate now to handle cockroach outbreaks.

**Comment by Aurx** [comment]
@SonixApache
Beecause bees are the heroes we deserve.

**Comment by Dacendeth** [comment]
Give bees ai to ventcrawl like scp 173 where it randomly teleports from vent to vent looking for people to murder.

**Comment by Pathid** [comment]
#### To do:
- [x] Working ventcrawling
- [x] Player-controlled eating
- [x] Actually targeting roach eggs

File: code/modules/mob/living/simple_animal/friendly/lizard.dm [hunk 3]
```diff
@@ -18,3 +18,61 @@
...
+
+/mob/living/simple_animal/hostile/lizard/proc/gulp(var/atom/eat_this)
+	visible_message("\The [name] consumes [eat_this] in a single gulp.", "<span class='notice'>You consume [eat_this] in a single gulp.</span>")
+	playsound(get_turf(src),'sound/items/egg_squash.ogg', rand(30,70), 1)
+	qdel(eat_this)
+	adjustBruteLoss(-2)
+
+/mob/living/simple_animal/hostile/lizard/ListTargets()
+	var/list/L = new()
+	L.Add(oview(vision_range, src))
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/friendly/lizard.dm]: Just `return oview(vision_range, src)`

File: code/modules/mob/living/simple_animal/friendly/lizard.dm [hunk 4]
```diff
@@ -18,3 +18,61 @@
 	mob_property_flags = MOB_NO_PETRIFY //Can't get petrified (nethack references)
 	meat_type = /obj/item/weapon/reagent_containers/food/snacks/meat/animal/lizard
 	held_items = list()
+
+	stop_automated_movement_when_pulled = TRUE
+	environment_smash_flags = FALSE
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/friendly/lizard.dm]: not a fan of using FALSE here, it's a bitflag, I'd use 0 if there's no more suitable define

File: code/modules/mob/living/simple_animal/friendly/lizard.dm [hunk 5]
```diff
@@ -18,3 +18,61 @@
...
+	pass_flags = PASSTABLE | PASSMOB
+	vision_range = 6
+	aggro_vision_range = 6
+	idle_vision_range = 6
+
+	var/static/list/edibles = list(/mob/living/simple_animal/cockroach, /obj/item/weapon/reagent_containers/food/snacks/roach_eggs) //Add bugs to this as they get added in
+
+/mob/living/simple_animal/hostile/lizard/proc/gulp(var/atom/eat_this)
+	visible_message("\The [name] consumes [eat_this] in a single gulp.", "<span class='notice'>You consume [eat_this] in a single gulp.</span>")
+	playsound(get_turf(src),'sound/items/egg_squash.ogg', rand(30,70), 1)
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/friendly/lizard.dm]: use `src` instead of `get_turf(src)`

**Comment by Pathid** [comment]
Could add bees but then they'd retaliate.

**Comment by Aurx** [comment]
You say that like it's a problem.

**Pull request merged**